src/test/java/org/ays/auth/model/enums/PermissionCategory.java

```diff
@@ -1,7 +1,58 @@
 package org.ays.auth.model.enums;
 
+import lombok.Getter;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Getter
 public enum PermissionCategory {
 
-    ROLE_MANAGEMENT
+    ROLE_MANAGEMENT(
+            Permission.ROLE_LIST,
+            Permission.ROLE_CREATE,
+            Permission.ROLE_UPDATE,
+            Permission.ROLE_DELETE,
+            Permission.ROLE_DETAIL
+    ),
+    PAGE(
+            Permission.INSTITUTION_PAGE,
+            Permission.LANDING_PAGE
+    ),
+    USER_MANAGEMENT(
+            Permission.USER_LIST,
+            Permission.USER_CREATE,
+            Permission.USER_UPDATE,
+            Permission.USER_DELETE,
+            Permission.USER_DETAIL
+    ),
+    EVACUATION_APPLICATION_MANAGEMENT(
+            Permission.APPLICATION_EVACUATION_LIST,
+            Permission.APPLICATION_EVACUATION_UPDATE,
+            Permission.APPLICATION_EVACUATION_DETAIL
+    ),
+    REGISTRATION_APPLICATION_MANAGEMENT(
+            Permission.APPLICATION_REGISTRATION_LIST,
+            Permission.APPLICATION_REGISTRATION_CREATE,
+            Permission.APPLICATION_REGISTRATION_DETAIL,
+            Permission.APPLICATION_REGISTRATION_CONCLUDE
+    ),
+    SUPER_ADMIN(
+            Permission.SUPER
+    );
+
+    private final Permission[] permissions;
+
+    PermissionCategory(Permission... permissions) {
+        this.permissions = permissions;
+    }
```

Comment:
Lombok anotasyonu kullanınca bu metodu elde edemiyor muyuz? `@RequiredArgsConstructor` gibi

Reply:
Evet denedim ama hata aldım ve çözemedim, bu yüzden elle yazdım. Lombok, sınıf içinde final olan argümanları kabul ediyor ancak biz başka bir enum sınıfını argüman olarak gönderdiğimiz için sorun yaşandığını düşünüyorum.

Reply:
anladım tamamdır, eline sağlık 🙌🏼